Controller/Webhook/Url.php

```diff
@@ -103,21 +103,34 @@ public function execute(): ?ResultInterface
             ]);
         }
 
+        $stores = $this->storeManager->getStores();
         $api = $this->apiFactory->create();
         foreach ($documentIds as $documentId) {
             $document = $api->getByID($documentId);
             if (!$document) {
                 continue;
             }
 
-            $urlRewrite = $this->findUrlRewrite($document, $store);
+            $currentStore = null;
+            foreach ($stores as $store) {
+                if ($this->configuration->getContentLanguage($store) === $document->lang) {
+                    $currentStore = $store;
+                    break;
+                }
+            }
+
+            if (!$currentStore) {
+                continue;
+            }
```

Comment:
Same code as in `Controller/Webhook/Unpublished.php`. Way nicer to use an function for it